frontend/packages/ceph-storage-plugin/src/components/modals/volume-snapshot-modal/volume-snapshot-modal.tsx

```diff
@@ -0,0 +1,116 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import {
+  Dropdown,
+  HandlePromiseProps,
+  withHandlePromise,
+} from '@console/internal/components/utils';
+import { Form, FormGroup, TextInput } from '@patternfly/react-core';
+import { K8sResourceKind, k8sCreate } from '@console/internal/module/k8s';
+import {
+  ModalBody,
+  ModalComponentProps,
+  ModalSubmitFooter,
+  ModalTitle,
+  createModalLauncher,
+} from '@console/internal/components/factory';
+import { VolumeSnapshotModel } from '../../../models';
+
+export const VolumeSnapshotModal = withHandlePromise((props: VolumeSnapshotModalProps) => {
+  const { close, cancel, resource, errorMessage, inProgress } = props;
+  const [buttonDisabled, setButton] = React.useState(false);
+  const [snapshotName, setName] = React.useState(
+    `${`${_.get(props.resource, 'metadata.name')}-snapshot-1`}`,
+  );
+  const snapshotTypes = {
+    single: 'Single Snapshot',
+  };
+
+  // To be extended
+  // const [selectedVersion, setSnapshotType] = React.useState(snapshotTypes.single)
+
+  // const handleTypeChange = (obj) => {
+  //     setSnapshotType(obj.value);
+  // }
+  const onValueChange = (value: string) => {
+    if (value === '') {
+      setButton(true);
+    } else {
+      setButton(false);
+    }
+    setName(value);
+  };
+
+  const submit = (event: React.FormEvent<EventTarget>) => {
+    event.preventDefault();
+    const snapshotTemplate: K8sResourceKind = {
+      apiVersion: 'snapshot.storage.k8s.io/v1alpha1',
+      kind: 'VolumeSnapshot',
+      metadata: {
+        name: `${snapshotName}`,
+        namespace: `${_.get(props.resource, 'metadata.namespace')}`,
```

Comment:
Avoid `_.get`. Use optional chaining instead. But if `resource` is never undefined, just use `resource.metadata.namespace`